### PR TITLE
test(integration-tests): simplify integration tests

### PR DESCRIPTION
**Pull request by armano2**
Goal of this PR is simplify and speed up test run time. 

Setting up docker container in this case seems to be overkill that takes ~7min on CI to execute, (all other tests are executed in ~4min)

This change removes ~7min out of CI run

Notable changes: 
- docker is dropped from integration tests in favor of executing isolated commands.
- integration tests are executed on all specified node versions

### CODE REVIEW

File: packages/integration-tests/tests/integration.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  it('it should produce the expected lint output', () => {
+    expect(runEslint('vue-jsx', '*.vue')).toMatchSnapshot();
+  });
+});
+
+describe('vue-sfc', () => {
+  it('it should produce the expected lint output', () => {
+    expect(runEslint('vue-sfc', '*.vue')).toMatchSnapshot();
+  });
+});
```
Comment (bradzacher) [inline packages/integration-tests/tests/integration.ts]: can this just be a map with a loop, like

```ts
const integrationTests = {
  'vue-jsx': '*.vue',
  // ....
};

Object.entries(integrationTests).forEach(([folderName, extension])=> {
  describe(folderName, () => {
    it('it should produce the expected lint output', () => {
      expect(runEslint(folderName, extension)).toMatchSnapshot();
    });
  });
});
```

File: packages/integration-tests/tests/integration.ts [hunk 2]
```diff
@@ -37,36 +37,16 @@ function runEslint(directory: string, paths: string): unknown {
...
-describe('markdown', () => {
+const integrationTests: [string, string][] = [
+  ['markdown', '*.md'],
+  ['recommended-does-not-require-program', '*.ts'],
+  ['typescript-and-tslint-plugins-together', '*.ts'],
+  ['vue-jsx', '*.vue'],
+  ['vue-sfc', '*.vue'],
+];
+
+describe.each(integrationTests)('%s', (directory, path) => {
```
Comment (bradzacher) [inline packages/integration-tests/tests/integration.ts]: image

**Comment by bradzacher** [comment]
I'm okay with this change, as it reduces friction for people working with the integration tests, and improves the CI feedback cycle.

I'd like to hear from @JamesHenry as to why he chose to use docker for the tests, so we can evaluate if dropping it is the best choice.

**Comment by armano2** [comment]
i'm noticed that sometimes current integration tests do fail on linking plugins

https://dev.azure.com/typescript-eslint/TypeScript%20ESLint/_build/results?buildId=3178&view=logs&j=18ce2c0c-8e5f-561b-2a1a-c5027ed6632f&t=dc4e626b-6a70-5458-4122-10e3d298b548
```
 npx: installed 123 in 4.629s
recommended-does-not-require-program      | Failed to load plugin '@typescript-eslint' declared in '--config': Cannot find module 'eslint'
recommended-does-not-require-program      | Referenced from: /usr/linked/.eslintrc.yml
recommended-does-not-require-program      | npx: installed 487 in 12.65s
recommended-does-not-require-program      | FAIL ./test.js
recommended-does-not-require-program      |   ● Test suite failed to run
```

**Comment by armano2** [comment]
@bradzacher recently there was issues with integration tests related to invalid versions: https://github.com/typescript-eslint/typescript-eslint/runs/436798850

**Comment by armano2** [comment]
lerna error seems to be related to https://status.npmjs.org/incidents/lvgmcrfvrvsh

**Comment by bradzacher** [comment]
Yeah I noticed the failures. With how frequently they've been occurring, I'm becoming strongly in favour of removing docker.

The only problem I have with this approach is that it'll pollute the project's node_modules in dev. Not sure if this is a problem or not.

**Comment by JamesHenry** [comment]
Yes, isolation and determinism were the reasons behind going with docker. If that is not playing out in practice then I am definitely happy for the docker orchestration to be removed

**Comment by bradzacher** [comment]
I'm going to close this for now. We can revisit this later if we want to, but for now this branch is way out of date.

**Pull request closed without merging**